core/src/main/java/com/amplitude/core/utilities/EventsFileManager.kt

```diff
@@ -165,10 +166,12 @@ class EventsFileManager(
 
     // write to underlying file
     private fun writeToFile(content: ByteArray, file: File) {
-        FileOutputStream(file, true).use {
-            it.write(content)
-            it.flush()
-        }
+        try {
+            FileOutputStream(file, true).use {
+                it.write(content)
+                it.flush()
+            }
+        } catch (e: FileNotFoundException) {}
```

Comment:
When would this happen actually? Race condition for writing and deleting? Any thoughts on if we could recover from the issue? 

Another thought is for this kind of exception we could do some debug logging?

Reply:
I think file not found exceptions are happening when file content was sent out by another coroutine and deleted. This write operation are from rollover() -> finish() -> write a ']' to close the array in the file. Maybe try catch exception from the EventPipeline.update() where called storage.rollover() is better.

Reply:
OK, I guess as long as we could do a retry sounds good. Maybe better to mimic in a test case.